skimage/metrics/set_metrics.py

```diff
@@ -109,8 +113,69 @@ def hausdorff_pair(image0, image1):
     max_dist_from_b = nearest_dists_from_a[max_index_from_b]
 
     if max_dist_from_b > max_dist_from_a:
-        return a_points[max_index_from_b], \
-            b_points[nearest_b_point_indices_from_a[max_index_from_b]]
+        return (
+            a_points[max_index_from_b],
+            b_points[nearest_b_point_indices_from_a[max_index_from_b]],
+        )
     else:
-        return a_points[nearest_a_point_indices_from_b[max_index_from_a]], \
-            b_points[max_index_from_a]
+        return (
+            a_points[nearest_a_point_indices_from_b[max_index_from_a]],
+            b_points[max_index_from_a],
+        )
+
+
+def modified_hausdorff_distance(image0, image1):
+    """Calculates the Modified Hausdorff distance between nonzero elements of
+    given images.
+
+    The Modified Hausdorff Distance (MHD) has been shown to perform better
+    than the directed Hausdorff Distance (HD) in the following work by
+    Dubuisson et al. [1]_. The function calculates forward and backward
+    mean distances and returns the largest of the two.
+
+    Parameters
+    ----------
+    image0, image1 : ndarray
+        Arrays where ``True`` represents a point that is included in a
+        set of points. Both arrays must have the same shape.
+
+    Returns
+    -------
+    distance : float
+        The Modified Hausdorff distance between coordinates of nonzero pixels in
+        ``image0`` and ``image1``, using the Euclidian distance.
+
+    References
+    ----------
+    .. [1] M. P. Dubuisson and A. K. Jain. A Modified Hausdorff distance for object
+       matching. In ICPR94, pages A:566-568, Jerusalem, Israel, 1994.
+       http://ieeexplore.ieee.org/xpls/abs_all.jsp?arnumber=576361
```

Comment:
```suggestion
       :DOI:`10.1109/ICPR.1994.576361`
       http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.1.8155
```
This DOI tag will get converted to a link to the IEEE article on document generation. I also added a non-paywalled link via CiteSeerX